components/script/dom/xmlhttprequest.rs

```diff
@@ -676,12 +676,12 @@ impl XMLHttpRequestMethods for XMLHttpRequest {
     fn SetResponseType(&self, response_type: XMLHttpRequestResponseType) -> ErrorResult {
         match self.global() {
             GlobalRoot::Worker(_) if response_type == XMLHttpRequestResponseType::Document
-            => return Ok(()),
+                => return Ok(()),
+            GlobalRoot::Window(_) if self.sync.get() => return Err(Error::InvalidAccess),
```

Comment:
I suspect that the test failures are due to the re-ordering of steps. The match-arm for `GlobalRoot::Worker(_)` above is Step 1, whereas the match-arm for `GlobalRoot::Window(_) if self.sync.get()` is Step 3, and down there the match for `self.ready_state.get()` is Step 2.